docs/source/daml/intro/6_Parties.rst

```diff
@@ -0,0 +1,197 @@
+
+To allow an ``iou.owner`` to create such a proposal, you need to give them the choice to propose a transfer on the ``Iou`` contract. The choice looks just like the above ``Transfer`` choice, except that a ``IouTransferProposal`` is created instead of an ``Iou``:
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- PROPOSE_TRANSFER_BEGIN
+  :end-before: -- PROPOSE_TRANSFER_END
+
+Bob can now transfer his ``Iou``. The transfer workflow can even be used for issuance:
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- IOU_TRANSFER_SCENARIO_BEGIN
+  :end-before: -- IOU_TRANSFER_SCENARIO_END
+
+Use role contracts for ongoing authorization
+--------------------------------------------
+
+Many actions, like the issuance of assets or their transfer, can be pre-agreed. You can represent this succinctly in DAML through relationship or role contracts.
+
+Jointly, an ``owner`` and ``newOwner`` can transfer an asset, as demonstrated in the scenrario above. In :doc:`7_Composing`, you will see how to compose the ``ProposeTransfer`` and ``IouTransferProposal_Accept`` choices into a single new choice, but for now, here is a different way. You can give them the joint right to transfer an iou:
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- MUTUAL_TRANSFER_BEGIN
+  :end-before: -- MUTUAL_TRANSFER_END
+
+Up to now, the controllers of choices were known from the current contract. Here, the ``newOwner`` variable is part of the choice arguments, not the ``Iou``. The above syntax is an alternative to ``controller c can``, which allows for this. Such choices live outside any ``controller c can`` block. They declared using the ``choice`` keyword, and have an extra clause ``controller c``, which takes the place of ``controller c can``, and has access to to the choice arguments.
+
+This is also the first time we have shown a choice with more than one controller. If multiple controllers are specified, the authority of *all* the controllers is needed. Here, neither ``owner``, nor ``newOwner`` can execute a transfer unilaterally, hence the name ``Mutual_Transfer``.
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- SENDER_ROLE_BEGIN
+  :end-before: -- SENDER_ROLE_END
+
+The above ``IouSender`` contract now gives a one party, the ``sender`` the right to send ``Iou`` contracts with positive amounts to a ``receiver``. The ``nonconsuming`` keyword on the choice ``Send_Iou`` changes the behaviour of the choice so that the contract it's exercised on does not get archived when the choice is exercised. That way the ``sender`` can use the contract to send multiple Ious. Here it is in action:
```

Comment:
```suggestion
The above ``IouSender`` contract now gives a one party, the ``sender`` the right to send ``Iou`` contracts with positive amounts to a ``receiver``. The ``nonconsuming`` keyword on the choice ``Send_Iou`` changes the behaviour of the choice so that the contract it's exercised on does not get archived when the choice is exercised. That way the ``sender`` can use the contract to send multiple Ious. 

Here it is in action:
```